python/python/bystro/api/auth.py

```diff
@@ -0,0 +1,370 @@
+        The authentication state, or None if the state file doesn't exist.
+    """
+    if os.path.exists(CREDENTIALS_PATH):
+        with open(CREDENTIALS_PATH, "r", encoding="utf-8") as f:
+            data = f.read()
+
+            if not data:
+                return None
+
+            return mjson.decode(data, type=CachedAuth)
+
+    return None
+
+
+def save_state(data: CachedAuth, print_result=True) -> None:
+    """
+    Saves the authentication state to a file.
+
+    Parameters
+    ----------
+    data : CachedAuth
+        The data to save.
+    print_result : bool, optional
+        Whether to print the result of the save operation, by default True.
+
+    Returns
+    --------
+    None
+    """
+    encoded_data = mjson.encode(data).decode("utf-8")
+
+    with open(CREDENTIALS_PATH, "w", encoding="utf-8") as f:
+        f.write(encoded_data)
+
+    if print_result:
+        print(
+            f"\nSaved auth credentials to {CREDENTIALS_PATH}:\n{mjson.format(encoded_data, indent=4)}"
+        )
+
+
+def signup(
+        email: str,
+        password: str,
+        name: str,
+        host: str,
+        port: int | None = None,
+        print_result=True
+    ) -> CachedAuth:
+    """
+    Signs up for Bystro with the given email, name, and password. Additionally, logs in and
+    saves the credentials, to enable API calls without re-authenticating.
+
+    Parameters
+    ----------
+    email : str
+        The email address for the account.
+    name : str
+        The name of the user.
+    password : str
+        The password for the account.
+    host : str
+        The hostname or IP address of the Bystro server.
+    port : int | None
+        (Optional) The port number on which the Bystro server is listening.
+    print_result : bool, optional
+        Whether to print the result of the signup operation, by default True.
+
+    Returns
+    -------
+    CachedAuth
+        The cached authentication state.
+    """
+    if os.path.exists(CREDENTIALS_PATH):
+        print("Existing session found, logging out")
+        logout()
+
+    if print_result:
+        print(f"\nSigning up for Bystro with email: {email}, name: {name}")
+
+    fq_host = _fq_host(host, port)
+    url = f"{fq_host}/api/user"
+
+    data = {"email": email, "name": name, "password": password}
+
+    response = requests.put(url, data=data, timeout=30)
+
+    if response.status_code != 200:
+        raise RuntimeError(
+            f"Login failed with response status: {response.status_code}. Error: \n{response.text}\n"
+        )
+
+    res = mjson.decode(response.text, type=SignupResponse)
+    state = CachedAuth(
+        access_token=res.access_token,
+        url=fq_host,
+        email=email,
+    )
+
+    save_state(
+        state,
+        print_result,
+    )
+
+    if print_result:
+        print("\nSignup & authentication successful. You may now use the Bystro API!\n")
+
+    return state
+
+
+def login(
+        email: str,
```

Comment:
These arguments seems overly indented ( 8 spaces). If you run through VS code's Format Document (command + shift + P), using black, on my machine I get 4 spaces.

Same for `def signup`.

Not a huge deal, but we should be consistent. In api/annotate.py we have 4 lines indent for `def create_jobs`